### PR TITLE
TensorBoard Override for same Repository

### PR DESCRIPTION
**Pull request by merveenoyan**
TensorBoard override enabled for when user wants to push to same repository

### CODE REVIEW

**Comment by merveenoyan** [comment]
I ran the style on my local it's so weird it's not passing

File: tests/test_keras_integration.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ def __init__(self, **kwargs):
         def call(self, x):
             return self.l1(x)
 
+
```
Comment (osanseviero) [inline tests/test_keras_integration.py:64]: nit: remove empty line

File: tests/test_keras_integration.py [hunk 2]
```diff
@@ -324,6 +325,53 @@ def test_push_to_hub_tensorboard(self):
...
+            git_email="ci@dummy.com",
+        )
+
+        os.remove(f"{WORKING_REPO_DIR}/tb_log_dir/tensorboard.txt")
+        with open(f"{WORKING_REPO_DIR}/tb_log_dir/override.txt", "w") as fp:
+            fp.write("Keras FTW")
+
+        REPO_NAME = repo_name("PUSH_TO_HUB")
+        model = self.model_init()
+        model.build((None, 2))
```
Comment (osanseviero) [inline tests/test_keras_integration.py:352]: This is already defined, I don't think there's a need to re-define

File: tests/test_keras_integration.py [hunk 3]
```diff
@@ -324,6 +325,53 @@ def test_push_to_hub_tensorboard(self):
...
+            api_endpoint=ENDPOINT_STAGING,
+            use_auth_token=self._token,
+            git_user="ci",
+            git_email="ci@dummy.com",
+        )
+
+        os.remove(f"{WORKING_REPO_DIR}/tb_log_dir/tensorboard.txt")
+        with open(f"{WORKING_REPO_DIR}/tb_log_dir/override.txt", "w") as fp:
+            fp.write("Keras FTW")
+
```
Comment (osanseviero) [inline tests/test_keras_integration.py:349]: Some idea to keep things simpler
```suggestion
        with open(f"{WORKING_REPO_DIR}/tb_log_dir_2/override.txt", "w") as fp:
            fp.write("Keras FTW")

```

File: tests/test_keras_integration.py [hunk 4]
```diff
@@ -324,6 +325,53 @@ def test_push_to_hub_tensorboard(self):
...
+        REPO_NAME = repo_name("PUSH_TO_HUB")
+        model = self.model_init()
+        model.build((None, 2))
+        push_to_hub_keras(
+            model,
+            repo_path_or_name=f"{WORKING_REPO_DIR}/{REPO_NAME}",
+            log_dir=f"{WORKING_REPO_DIR}/tb_log_dir",
```
Comment (osanseviero) [inline tests/test_keras_integration.py:356]:
```suggestion
            log_dir=f"{WORKING_REPO_DIR}/tb_log_dir_2",
```

File: tests/test_keras_integration.py [hunk 5]
```diff
@@ -324,6 +325,53 @@ def test_push_to_hub_tensorboard(self):
...
+
+        model_info = HfApi(endpoint=ENDPOINT_STAGING).model_info(
+            f"{USER}/{REPO_NAME}",
+        )
+        self.assertTrue(
+            "logs/override.txt" in [f.rfilename for f in model_info.siblings]
+        )
+        self.assertFalse(
+            "logs/tensorboard.txt" in [f.rfilename for f in model_info.siblings]
+        )
```
Comment (osanseviero) [inline tests/test_keras_integration.py:371]: Very cool!

**Review by osanseviero — COMMENTED** [review]
Great 🔥 Thanks for the PR. The PR looks good, I left minor nits and pretty much ready to merge

File: tests/test_keras_integration.py (hunk 4 above)
Comment (merveenoyan) [reply tests/test_keras_integration.py:351]: ok I'll remove the removal of the file and have a separate directory instead

File: tests/test_keras_integration.py (hunk 3 above)
Comment (merveenoyan) [reply tests/test_keras_integration.py]: I thought it would be less hassle 😂

**Review by osanseviero — APPROVED** [review]
Looks great! Thanks for the PR

**Pull request merged**